### DISCUSSION TITLE
GPS elevation correction based on real geoid (sea level)

### DISCUSSION THREAD

**Discussion by alccrl** [iNavFlight/inav#7304] [Q&A]
Hi,

does Inav (3.0.1 as reference) calculate altitude corrections when showing/using sea level altitudes ?
(Gps altitude reading is often automaticly corrected while using smartphone outdoors apps).

I had a fast look at inav source code, wiki, configurator and cli, but I did not find anything about.
Planning inav missions (they work, very nice feature), as example in mountain environment, it is important to  accurately forecast altitudes, in order not to hit rocks or to fly higher than necessary. 

Does inav take into account these elevation differences ?

Alfredo

**Comment by stronnag** [comment]
inav missions can be defined with with relative (to arming) or absolute (above the WGS84 geoid) elevations. 

So I don't understand what you're asking. Please try and clarify.

**Reply by alccrl** [reply]
I'll try to better explain ...

Gps native altitude NMEA output is not corrected for geoid inaccuracies (local earth differences against elipsoide).
These differences may be more than 50 meters, and many smartphone apps provide means to correct these for outdoors purposes.
As an example : if I plan a waypoint to 2000 m sea level, and mountain is 1990 m sea level at that point, I have 10 m margin.
If gps local (for that long/lat) elevation error is greater than 10 m, I much probably hit the mountain top during mission.

This topic is not only sw, it is also "geographic/topographic". :-)

Hope to have clarified ...

Alfredo

**Comment by stronnag** [comment]
Ultimately all inav elevations are relative to launch; absolute or relative WP merely change the datum is established.

* GPS vertical measurement is not so accurate
* You are using the cheapest consumer grade baro

It is necessary to the pilot to  plan within the constraints of the hardware. End of story.

**Reply by alccrl** [reply]
Thanks !
Anyway this was only a question, not a request for new features, because altitude correction would need an external data source, or internal inav look up tables, I guess, so not so simple to be implemented.

Bye

**Comment by stronnag** [comment]
Indeed, there exists things like [mwp's mission / terrain analysis](https://github.com/stronnag/mwptools/wiki/Mission-Elevation-Plot-and-Terrain-Analysis) which relies on an external terrain DB (which also provides elevations from the WGS84 geoid). But it's just a guide.

**Reply by avsaase** [reply]
Doesn't the new mission planner in the configurator also support altitude offsets? Or has that not been merged yet?

**Reply by alccrl** [reply]
Are You referring [Mission Planner] Elevation Profile & Home #1299 ?

**Reply by avsaase** [reply]
@alccrl I assume you meant https://github.com/iNavFlight/inav-configurator/pull/1299. Yes, that's what I referred to.
